src/common/com/intellij/plugins/haxe/ide/HaxeReadWriteAccessDetector.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.ide;
+
+import com.intellij.codeInsight.highlighting.ReadWriteAccessDetector;
+import com.intellij.plugins.haxe.lang.psi.*;
+import com.intellij.plugins.haxe.lang.util.HaxePsiUtil;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiReference;
+import org.jetbrains.annotations.NotNull;
+
+public class HaxeReadWriteAccessDetector extends ReadWriteAccessDetector {
+  @Override
+  public boolean isReadWriteAccessible(@NotNull final PsiElement element) {
+    return element instanceof HaxePsiField || element instanceof HaxeClass || element instanceof HaxeMethod;
```

Comment:
It's definitely will for local variables, but about array access I'm not sure

Reply:
The trouble with array is: Even if the declaration states that the array itself is not write accessible, it's elements _are_.

Reply:
`arr[0] = 12;` is recognizing as read operation
`arr = [];` as write operation

It works in the same way as it java does. Should we make it recognize `arr[0] = 12;` as write operation?

Reply:
I think you're right. No need to make it a write operation.

Reply:
Is there a place for both?  I get that arr[0] *is* really just an array access.  It's also nice to be able to see that the array _contents_ are being modified.  But then again, if somebody takes a copy of the entry and modifies it, it would be very difficult for us to track that change, too.
 ```Haxe
  var myobj = arr[0];
  myobj.setSomeValue(0);  // Hard to follow that back to the read and mark that as read/write.
```